### ISSUE TITLE
Ensure cleanup is possible in case of Pod termination

### ISSUE THREAD

**Issue by Tomasz-Smelcerz-SAP** [kyma-project/test-infra#209]
Some existing prow jobs allocate external resources (clusters, VMs, disks etc) that must be cleaned up ASAP to reduce costs.
When a Pod dies or is explicitly terminated by Prow policy (few subsequent commits in a short period of time), how can we react to ensure resources are cleaned up?
If we can't react to Pod termination, our options might be:
- Put a label on created resources in order to clean them up later
- Put info about created resources into some kind of database to clean them up later
see also #177

Acceptance Criteria:
- Verify that current Prow policy terminates Pods if there are few commits in a short period of time and if we can somehow control the termination process.
- Verify if we can somehow capture Pod termination signal inside the Pod and react upon it.

**Comment by Demonsthere** [comment]
I think we might be able to omit the signal handling problem by focusing not on container termination, but rather pod lifecycle. We might want to consider a [preStop](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#container-hooks) signal which would run always when pod is terminated. I think this might a good place to run our cleanup scripts.

**Comment by Demonsthere** [comment]
Some additional piece of [documentation](https://unofficial-kubernetes.readthedocs.io/en/latest/concepts/containers/container-lifecycle-hooks/), from which the most important for us part would be:
> PreStop
> This hook is called immediately before a container is terminated. No parameters are passed to the handler. This event handler is blocking, and must complete before the call to delete the container is sent to the Docker daemon. The SIGTERM notification sent by Docker is also still sent. A more complete description of termination behavior can be found in Termination of Pods.

**Comment by aszecowka** [comment]
@Demonsthere @Tomasz-Smelcerz-SAP generally it is a good idea, but I am wondering if this limitation can be a problem for us:
> Users should make their hook handlers as lightweight as possible

As we discussed with @Tomasz-Smelcerz-SAP , this task has low priority and now please focus on implementing period jobs that performs cleanup.

**Comment by aszecowka** [comment]
Closing issue after discussion with @Tomasz-Smelcerz-SAP - periodic jobs should be enough
